### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by degisew**
### OOP school library: set up associations
**In this PR, I implemented the following features:**
- Created a class called Classroom with the following:
          @label instance variable should be initialized in the constructor.
          Setter and getter for @label.
- Created the has-many/belongs-to relationship between Classroom and Student. 
- Create da class Book with the following:
          @title and @author instance variables, should be initialized in the constructor.
          Setters and getters for instance variables.
- Created a class Rental with the following:
         @date instance variable should be initialized in the constructor.
         Setter and getter for @date
- Created the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :author, :title
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (elisha2kyakpo1) [inline book.rb]: - [ ] A book class should have a method for adding a rental.

```
 def add_rental(person, date)
    # *the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. 
    #code here...
 end
```

File: person.rb [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
   attr_accessor :name, :age
 
   def initialize(age, name = 'unknown', parent_permision: true)
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permision
+    @rentals = []
     super()
   end
```
Comment (elisha2kyakpo1) [inline person.rb:15]: - [ ]   A method for adding rentals should be present.
  ```ruby
  def add_rental(book, date)
    # *The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association
    #*Code here...*
  end
```

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @degisew, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ The PR title and description look good and professional.
✔️ No linter errors.
✔️ Descriptive readme file.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (degisew) [reply person.rb:15]: Okay, what a great view! Thanks a lot. I will walk through it

**Review by jasonfelice — APPROVED** [review]
Hi @degisew, 

Great job implementing the changes required by the previous reviewer! 👍🏼 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<p align="center"></p>

## Highlights
No Linter errors✔️
Professional PR✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**